integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java

```diff
@@ -373,19 +388,34 @@ private static void usage() {
   }
 
   private static List<CommandInfo.URI> getExecutorDependencyURIList() {
-    String dependencyPath =
-        Configuration.get(PropertyKey.INTEGRATION_MESOS_EXECUTOR_DEPENDENCY_PATH);
-    return Lists.newArrayList(
-        CommandInfo.URI.newBuilder()
-            .setValue(PathUtils.concatPath(dependencyPath, "alluxio.tar.gz")).setExtract(true)
-            .build(), CommandInfo.URI.newBuilder()
-            .setValue(Configuration.get(PropertyKey.INTEGRATION_MESOS_JRE_URL)).setExtract(true)
-            .build());
+    List<URI> dependencies = new ArrayList<>();
+    if (installJavaFromUrl()) {
+      dependencies.add(CommandInfo.URI.newBuilder()
+          .setValue(Configuration.get(PropertyKey.INTEGRATION_MESOS_JRE_URL)).setExtract(true)
+          .build());
+    }
+    if (installAlluxioFromUrl()) {
+      dependencies.add(CommandInfo.URI.newBuilder()
+          .setValue(Configuration.get(PropertyKey.INTEGRATION_MESOS_ALLUXIO_JAR_URL))
+          .setExtract(true).build());
+    }
+    return dependencies;
   }
 
-  private static Protos.Credential createCredential() {
+  private static boolean installJavaFromUrl() {
+    return Configuration.containsKey(PropertyKey.INTEGRATION_MESOS_JRE_URL) && !Configuration
+        .get(PropertyKey.INTEGRATION_MESOS_JRE_URL).equalsIgnoreCase("PREINSTALLED");
```

Comment:
nit: I would prefer to use "LOCAL" instead of "PREINSTALLED" and we should also create a named constant for this to avoid duplication